xbmc/platform/android/CPUInfoAndroid.cpp

```diff
@@ -8,17 +8,43 @@
 
 #include "CPUInfoAndroid.h"
 
+#include "URL.h"
+#include "utils/StringUtils.h"
 #include "utils/Temperature.h"
 
 #include "platform/android/activity/AndroidFeatures.h"
 
+#include <vector>
+
 std::shared_ptr<CCPUInfo> CCPUInfo::GetCPUInfo()
 {
   return std::make_shared<CCPUInfoAndroid>();
 }
 
 CCPUInfoAndroid::CCPUInfoAndroid()
 {
+  m_pPosixFile = std::make_unique<CPosixFile>();
+
+  if (m_pPosixFile && m_pPosixFile->Open(CURL("/proc/cpuinfo")))
+  {
+    std::vector<char> buffer(2048, 0);
```

Comment:
just out of curiosity why an `std::vector` instead of an `std::array` if this seems to have a fixed size and is not dynamically resized? Wouldn't a `std::array` fit better this use case, although performance improvements are pretty much negligible?

Reply:
I agree. For small buffers it is better (and more logical) to use the stack. Vectors are allocated on heap is no reason for use it if amount of memory is very small and fixed.